sklearn/linear_model/stochastic_gradient.py

```diff
@@ -426,21 +441,27 @@ def _fit(self, X, y, alpha, C, loss, learning_rate, coef_init=None,
         # Clear iteration count for multiple call to fit.
         self.t_ = None
 
-        self._partial_fit(X, y, alpha, C, loss, learning_rate, self.n_iter,
+        self._partial_fit(X, y, alpha, C, loss, learning_rate, self.max_iter,
                           classes, sample_weight, coef_init, intercept_init)
 
+        if self.n_iter_ == self.max_iter:
```

Comment:
Please change this test to:

```python
if self.tol > 0 and self.n_iter_ == self.max_iter:
```

so that the user can disable the `ConvergenceWarning` when he/she decide to always perform `max_iter` iterations intentionally (effectively disabling the stopping criterion).